module/Finna/src/Finna/View/Helper/Root/Record.php

```diff
@@ -1285,4 +1308,80 @@ public function getOrganisationMenuPosition()
         return ('SolrAuth' === $source || $this->hasLargeImageLayout())
             ? 'inline' : 'sidebar';
     }
+
+    /**
+     * Returns HTML for encapsulated records.
+     *
+     * @param array $opt        Options
+     * @param ?int  $offset     Record offset
+     *                          (used when loading more results via AJAX)
+     * @param ?int  $indexStart Result item offset in DOM
+     *                          (used when loading more results via AJAX)
+     *
+     * @return string
+     */
+    public function renderEncapsulatedRecords(
+        array $opt = [],
+        ?int $offset = null,
+        ?int $indexStart = null
+    ): string {
+        foreach (array_keys($opt) as $key) {
+            if (!in_array(
+                $key,
+                [
+                    'limit',
+                    'page',
+                    'showAllLink',
+                    'view',
+                ]
+            )
+            ) {
+                unset($opt[$key]);
+            }
+        }
+
+        $id = $opt['id'] = $this->driver->getUniqueID();
+
+        $loadMore = (int)$offset > 0;
+
+        // null is an accepted limit value (no limit)
+        if (!array_key_exists('limit', $opt)) {
+            $opt['limit'] = 6;
+        }
+        $opt['showAllLink'] = $opt['showAllLink'] ?? true;
+        $view = $opt['view'] = $opt['view'] ?? 'grid';
+
+        $resultsCopy = clone $this->encapsulatedResults;
```

Comment:
Voisiko tämän tehkä callback-metodilla, joka annetaan constructorissa? Silloin ei tarvitsisi luoda encapsulatedResults:ia etukäteen eikä muistaa `clone`:a. Tyyliin: `$resultsCopy = $this->getEncapsulatedResults(new Parameters($opt))`

Ja factoryn constructor-kutsussa:

```
            [...],
            $container->get(\VuFind\Form\Form::class),
            function ($params) {
                $result = clone $container->get(\VuFind\Search\Results\PluginManager::class)
                    ->get('EncapsulatedRecords');
                $result->initFromRequest($params);
                return $result;
            }
        );
```